lua/zk/pickers/minipick.lua

```diff
@@ -0,0 +1,70 @@
+local M = {}
+local H = {}
+local minipick = require("mini.pick")
+
+M.note_picker_list_api_selection = { "title", "path", "absPath" }
+
+M.show_note_picker = function(notes, opts, cb)
+  notes = vim.tbl_map(function(n)
+    return { text = n.title, path = n.absPath, note = n }
+  end, notes)
+
+  opts = opts or {}
+  local minipick_opts = vim.tbl_deep_extend("force", {
+    source = {
+      items = notes,
+      name = opts.title,
+      choose = function(_)
+        return nil
+      end,
+    },
+    window = {
+      prompt_prefix = opts.title .. "> ",
+    },
+  }, opts.minipick or {})
+
+  local item = minipick.start(minipick_opts)
+  if item ~= nil then
+    cb(opts.multi_select and { item.note } or item.note)
+  end
+end
+
+M.show_tag_picker = function(tags, opts, cb)
+  tags = vim.tbl_map(function(t)
+    local padded_cnt = H.ensure_text_width(t.note_count, opts.note_count_width or 4)
+    return {
+      text = string.format("%s │ %s", padded_cnt, t.name),
+      tag = t,
+    }
+  end, tags)
+
+  opts = opts or {}
+  local minipick_opts = vim.tbl_deep_extend("force", {
+    source = {
+      items = tags,
+      name = opts.title,
+      choose = function(_)
+        return nil
+      end,
+    },
+    window = {
+      prompt_prefix = opts.title .. "> ",
+    },
+  }, opts.minipick or {})
+
+  local item = minipick.start(minipick_opts)
+  if item ~= nil then
+    cb(opts.multi_select and { item.tag } or item.tag)
+  end
```

Comment:
does mini not support multi selection?

Reply:
I think it does. It's called "mark" in mini.pick.

Reply:
It does support multi-selection, but the `MiniPick.start()` method only returns a single note even if multiple notes were selected. Similar to the above comment about `choose`,`mini.pick` does have a `choose_marked` to take action on multiple selected items. For the same reasons above, I want to delegate to the user supplied callback function, but I don't know how to get access to the ... [scratch that last sentence, as I was typing this comment, I realized I should just be invoking the user-supplied callback in the `choose` functions. Let me play around with this and will update this weekend. Thanks for asking the question].